README.md

```diff
@@ -414,6 +414,51 @@ If you want to use git (without a submodule) then you can use `ExternalProject_A
 
 I recommend you use `ExternalProject_Add` instead of `add_subdirectory` for **Nabla** as we haven't  tested its use by *3rdparty* applications that use *CMake* to build themselves yet.
 
+# Caveats and Particular Behaviour
+
+## Hardcoded Caps
+
+### Max Descriptor Sets is always 4
+
+## Debugging with RenderDoc
+
+###  Non-programmatic OpenGL catpures will be delimited inconsistently
+
+Due to our no-pollution opengl state isolation policy, we have 1 queue or swapchain = 1 thread = 1 gl context + 1 master context and thread for device calls.
+
+Renderdoc therefore serializes all calls, and presents them inside the capture in interleaved order (records them on a single timeline "as they happened").
+
+Furthermore it has no idea what constitutes a frame, because swap-buffers call happens on a separate thread than all the other API calls. **So use the `IGPUQueue` start/end capture methods!**
+
+### RenderDoc flips images for display in the ImageViewer tab on OpenGL captures
+
+Ctrl+F `localRenderer in https://github.com/baldurk/renderdoc/blob/4103f6a5455b9734e9bf74e254577f5c03188136/renderdoc/core/image_viewer.cpp
+
+### OpenGL/Vulkan Inconsistacies
+
+In certain cases same calls to Vulkan and OpenGL might result in y-flipped image relevant to the other API.
+
+Both APIs write (-1,-1) in NDC space to (0,0) in image space (two wrongs make right), and memory-wise (0,0) always represents the lowest byte in memory.
+
+This inconsistency comes from swapchain presentation. When presenting the swapchain, the image location (0,0) corresponds to **bottom-left** in OpenGL and **top-left** in Vulkan.
+
+To account for this inconsistency, we expect the user to apply swapchain transforms. The `HORIZONTAL_MIRROR_180` swapchain transform corresponds to flipping Y, which, when done to the swapchain image, ensures the OpenGL output matches Vulkan. For this reason, the only swapchain transform exposed as supported by OpenGL is `HORIZONTAL_MIRROR_180`. To get the swapchain's transform, use [`ISwapchain getSurfaceTransform()`](https://github.com/Devsh-Graphics-Programming/Nabla/blob/aa31741485b7534d8c29ff2d55a2e498748fe223/include/nbl/builtin/glsl/utils/swapchain_transform.glsl#L64).
```

Comment:
@deprilula28 change this introduction text to the one I wrote below and
- [ ] change the function names when you apply them.
- [ ] provide links/update links for the functions (don't give permalinks because we might update our logic, just put the link to the file and they can just search and find it there. For example give link to ISwapchain.h file for ISwapchain::getSurfaceTransform)

#### Solution by Surface Transforms

We solve this inconsistency by using [surface transforms](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkSurfaceTransformFlagBitsKHR.html); This transforms are relative to `presentation engine’s natural orientation`. and we report `HORIZONTAL_MIRROR_180` support in our OpenGL backend and defer handling these rotations (relative to natural orientaion) to the user.

We provide helper functions in both glsl and C++ Nabla codebase to consider surface transforms, See [surface_transforms.glsl](Link)

Note that it is common to apply surface transformation to projection matrices to account for this fact. See [getSurfaceTransformationMatrix](Link) and [Android Developers Guide to Pre-rotation](https://developer.android.com/games/optimize/vulkan-prerotation)

Use [`ISwapchain getSurfaceTransform()`](link) to get the transformation from swapchain.